File: JavaScriptPractice/Leetcode53.js
var maxSubArray = function(nums) 
{
    let curMax = nums[0]
    let res= curMax
    if (nums.length==1) {
        return res;
    }
    for (let i = 1; i < nums.length; i++) {
        curMax = Math.max(curMax+nums[i],nums[i]);
        res=Math.max(res,curMax)
    }
    return res;
};

File: CSharpPractice/LeetcodePractice/Leetcode53.cs
public class Solution53 {
    public int MaxSubArray(int[] nums) 
    {
        int curMax =nums[0];
        int res = curMax;
        if (nums.Length==1)
        {
            return res;
        }
        for (int i = 1; i < nums.Length; i++)
        {
            curMax = Math.Max(curMax+nums[i],nums[i]);
            res =Math.Max(curMax,res);
        }
        return res;
    }
}